### ISSUE TITLE
Question: String template literal vs JSX compatibility?

### ISSUE THREAD

**Issue by kristoferjoseph** [developit/htm#77]
## *tl;dr* 
_edit_: **This issue is meant to be a broader discussion** based off of issues like: #56, #58, #60, #76

After upgrading to 2.* I ran into a lot of issues where htm removed functionality that string template literals ( and html 5 in general have ).

### So the question is: 
**Should htm be chasing jsx compatibility or string template literal behavior?**

## Issues that bit me
- no longer allowing tags that html 5 allows:
  - `<input type="text">` vs `<input type="text"/>`, `<img>` etc.
- no longer rendering nested template literals
```
<${MyComponent}
  href="/path/${id}"
><//>
```
now needs to be:
```
<${MyComponent}
  href="${'/path/' + id}"
><//>
```

These are small things that cause hard to debug issues for anyone familiar with how String Template Literals work.

**Comment by aciccarello** [comment]
I think this is a duplicate of #58

**Comment by kristoferjoseph** [comment]
@aciccarello not an exact duplicate, since it is more of a discussion, but yes thanks for mentioning that issue.

**Comment by developit** [comment]
Just a correction - the nested template literals thing isn't about nesting, it's just that we dropped support for mixed static + dynamic attribute value parts. Adding support back ends up being pretty cumbersome, and it was a feature that has no JSX analog. Also, it's worth noting that there is no performance benefit associated with mixed static+dynamic attribute values, whereas there is a real performance benefit associated with fully static attributes (as their resulting nodes can be cached).

```js
// this worked in HTM 1
html`<a href="/path/${id}" />`
// it compiled to this:
h('a', { href: "/path/" + id });

// this is what JSX requires:
<a href={"/path/" + id} />

// so in HTM 2 we simplified the parser to only have support for that:
html`<a href=${"/path/" + id} />`
```

**Comment by developit** [comment]
This was fixed by @jviide and released in HTM 2.2! 🎉
